components/brave_webtorrent/extension/manifest.json

```diff
@@ -7,6 +7,9 @@
     "scripts": ["extension/out/brave_webtorrent_background.bundle.js"],
     "persistent": true
   },
+  "web_accessible_resources": [
```

Comment:
@feross @diracdeltas WDYT about the following change?

I need it, because our internal network hook that redirects torrent links to the webtorrent extension is not working anymore with Network Service - such redirects become a subject for safety check, and check passes only for resources explicitly mentioned in the manifest.

Since this exposes `brave_webtorrent.html` to the outer world, I'd like to get some approval from security people.

Reply:
cc @feross since i'm not sure what all this file is capable of doing

being in web_accessible_resources means:
* sites can make an XHR to see if this file exists, which opens some fingerprinting risk (lets people tell that the browser is brave). IMO not a big deal.
* allows this resource to be accessed on external origins via CORS
* lets external origins navigate to this page
* maybe allows iframing of this resource?

if `brave_webtorrent.html` is just a static HTML page, this seems fine. if it includes any user-specific information or anything sensitive, i would be concerned about this

Reply:
It is the main webtorrent page that allows to start/stop torrent, shows progress, etc.

Basically before this change any resource can provoke showing this page it by issuing something looking like a torrent (mime, filename), since we automatically redirect such downloads to this page.

With this change it becomes directly accessible and, yes, iframable :(
![Screenshot from 2019-08-09 11-31-03](https://user-images.githubusercontent.com/3109556/62754308-82e02100-ba99-11e9-8825-9992c7219357.png)

Reply:
If sites can read the plain HTML of the page over XHR, that's fine. It's just a static file. Letting pages navigate to it is also fine, since as @iefremov says we already allow sites to navigate to `.torrent` files and magnet links. But we don't want to allow the page to be rendered in an iframe. The risk of clickjacking is too much.

What happens if you update the webtorrent extension's CSP to add `frame-ancestors 'none'`?

Reply:
much better!

> Refused to display 'chrome-extension://lgjmpdmojkpocjcopdikifhejkkjglho/extension/brave_webtorrent.html' in a frame because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'none'".

Reply:
Another approach would be to prevent iframe requests from being intercepted in the first place. You might be able to change this function https://github.com/brave/brave-core/blob/7e5a2525cebb90be4b924ad6cb0ff62d867cd5bf/components/brave_webtorrent/browser/net/brave_torrent_redirect_network_delegate_helper.cc#L78-L86

to return true for frames (i.e. a top level page) and return false for all other resource types. Currently it also returns true for subframes (i.e. a frame or iframe) which we don't really want to do.

With this change, the extension won't intercept the request and the site will just be iframing a literal .torrent file, which should just cause the file to get downloaded.

Reply:
implemented both suggestions, thanks a lot @feross  https://github.com/brave/brave-core/pull/2988/commits/7ffbcd1a5022ae8c9ff62092cdc2b048547136b6

@diracdeltas ok?

Reply:
@iefremov Nice, lgtm!

Reply:
sgtm